connectors/sources/sharepoint_online.py

```diff
@@ -1183,41 +1276,69 @@ def get_default_configuration(cls):
                 "order": 3,
                 "type": "str",
             },
+            "auth_method": {
+                "label": "Authentication Method",
+                "order": 4,
+                "type": "str",
+                "display": "dropdown",
+                "options": [
+                    {"label": "Client Secret", "value": "secret"},
+                    {"label": "Certificate", "value": "certificate"},
+                ],
+                "value": "certificate",
```

Comment:
I think this will be a breaking change, as on upgrade, we'll add this new config to existing connectors, right? But existing connectors will be using `secret`? I think we should default to `secret`, but tooltip that `certificate` is preferred?

Reply:
Yeah you're right, it should be "secret" for backwards compatibility!

Reply:
Fixed in https://github.com/elastic/connectors/pull/3064/commits/3365b740956fb5a23632499580851dfec63d4a37